From 2a299f3264e39fb0323601958725b8bbb5268a21 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Tue, 12 May 2020 20:55:27 +0300
Subject: [PATCH] [SW-0] ppboot/preboot: remove legacy code

We no longer use the legacy boot mechanisms, so remove them.

Change-Id: I889d7570d0d2f6354092c28512436d34983baeb6
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/app/ppboot.c b/app/ppboot.c
index dbf6c50..0fccf6a 100644
--- a/app/ppboot.c
+++ b/app/ppboot.c
@@ -18,37 +18,6 @@
 #include "spi.h"
 #include "pll.h"
 
-/* Stable ID can be 0 or 1, if not defined return primary offset (0) */
-static u32 next_image_base_offset(u32 offset, char *name)
-{
-	u32 id;
-
-	spi_copy(&id, offset, sizeof(id), name);
-	id = (id & 0xFF) - '1';
-
-	return (id == 1) ? HL_SPI_SECONDARY_BASE : HL_SPI_PRIMARY_BASE;
-}
-
-static enum fw_post_reset_cmd preboot_image_offset_legacy(u32 image_pending,
-							struct boot_desc *desc)
-{
-	u32 next_img_off = HL_SPI_NEXT_IMAGE_OFF;
-	char *name = "NEXT";
-
-	/* Check for normal boot or update pending */
-	if (image_pending == FLOPS_UPD_PENDING) {
-		next_img_off = HL_SPI_MODIFI_IMAGE_OFF;
-		name = "MODIFIABLE";
-	}
-
-	desc->spi_offset = next_image_base_offset(next_img_off, name) +
-						HL_SPI_LEGACY_PREBOOT_OFF;
-	desc->img_size = PREBOOT_SIZE;
-	desc->loadaddr = PREBOOT_TEXT_BASE;
-
-	return image_pending;
-}
-
 static enum fw_post_reset_cmd preboot_image_offset(struct boot_desc *desc)
 {
 	u32 image_pending = regmap_readl(mmUPD_PENDING_STS);
@@ -61,8 +30,14 @@
 			return image_pending;
 	}
 
-	preboot_puts("No boot descriptors detected, falling back to legacy\n");
-	return preboot_image_offset_legacy(image_pending, desc);
+	preboot_puts("No boot descriptors detected, need recovery!!!\n");
+	preboot_puts("Trying hard coded Preboot offsets (might hang)...\n");
+
+	desc->spi_offset = HL_SPI_LEGACY_PREBOOT_OFF;
+	desc->img_size = PREBOOT_SIZE;
+	desc->loadaddr = PREBOOT_TEXT_BASE;
+
+	return image_pending;
 }
 
 static u64 copy_preboot(void)
diff --git a/common/boot_desc.c b/common/boot_desc.c
index 9c81bc0..35db95d 100644
--- a/common/boot_desc.c
+++ b/common/boot_desc.c
@@ -15,7 +15,6 @@
 #include "uart.h"
 
 #define MAX_IMGS		5
-#define HL_FW_BOOT_PTRS_BASE	HL_SPI_NEXT_IMAGE_OFF
 
 #define BOOT_PTRS_SZ		sizeof(struct hl_fw_boot_ptrs)
 #define IMG_DESC_SZ		sizeof(struct hl_fw_img_desc)
diff --git a/gaudi/common.h b/gaudi/common.h
index 202a46e..b1e14d7 100644
--- a/gaudi/common.h
+++ b/gaudi/common.h
@@ -7,10 +7,8 @@
 
 #include <preboot.h>
 
-#define HL_SPI_NEXT_IMAGE_OFF		0x60000
-#define HL_SPI_MODIFI_IMAGE_OFF		0x60004
+#define HL_FW_BOOT_PTRS_BASE		0x60000
 #define HL_SPI_LEGACY_PREBOOT_OFF	0xA0000
-#define HL_SPI_LEGACY_UBOOT_OFF		0xB0000
 
 #define PPBOOT_TEXT_BASE		0x7FF0000000
 #define PREBOOT_TEXT_BASE		(PPBOOT_TEXT_BASE + PPBOOT_SIZE)
diff --git a/goya/common.h b/goya/common.h
index 0d42b4f..1500865 100644
--- a/goya/common.h
+++ b/goya/common.h
@@ -7,10 +7,8 @@
 
 #include <preboot.h>
 
-#define HL_SPI_NEXT_IMAGE_OFF		0x60000
-#define HL_SPI_MODIFI_IMAGE_OFF		0x60004
+#define HL_FW_BOOT_PTRS_BASE		0x60000
 #define HL_SPI_LEGACY_PREBOOT_OFF	0xA0000
-#define HL_SPI_LEGACY_UBOOT_OFF		0xB0000
 
 #define PPBOOT_TEXT_BASE	0x7FF0000000
 #define PREBOOT_TEXT_BASE	(PPBOOT_TEXT_BASE + PPBOOT_SIZE)
diff --git a/goya2/common.h b/goya2/common.h
index 26dfed2..36f6cf1 100644
--- a/goya2/common.h
+++ b/goya2/common.h
@@ -8,10 +8,8 @@
 #include <preboot.h>
 #include <goya2_blocks.h>
 
-#define HL_SPI_NEXT_IMAGE_OFF		0x80000
-#define HL_SPI_MODIFI_IMAGE_OFF		0x80004
+#define HL_FW_BOOT_PTRS_BASE		0x80000
 #define HL_SPI_LEGACY_PREBOOT_OFF	0xC0000
-#define HL_SPI_LEGACY_UBOOT_OFF		0xD0000
 
 #define PPBOOT_TEXT_BASE		0x7FF0000000
 #define PREBOOT_TEXT_BASE		(PPBOOT_TEXT_BASE + PPBOOT_SIZE)
diff --git a/goya2_arc_gnu/common.h b/goya2_arc_gnu/common.h
index f62ec56..d5968f9 100644
--- a/goya2_arc_gnu/common.h
+++ b/goya2_arc_gnu/common.h
@@ -8,10 +8,8 @@
 #include <preboot.h>
 #include <goya2_blocks_linux_driver.h>
 
-#define HL_SPI_NEXT_IMAGE_OFF		0x80000
-#define HL_SPI_MODIFI_IMAGE_OFF		0x80004
+#define HL_FW_BOOT_PTRS_BASE		0x80000
 #define HL_SPI_LEGACY_PREBOOT_OFF	0xC0000
-#define HL_SPI_LEGACY_UBOOT_OFF		0xD0000
 
 #define PCIE_FW_SRAM_BASE	0x1BFF0000
 #define FW_SRAM_BASE		0x1B050000
diff --git a/include/preboot.h b/include/preboot.h
index 6b56854..72e8b05 100644
--- a/include/preboot.h
+++ b/include/preboot.h
@@ -32,11 +32,8 @@
 #define PPBOOT_SIZE		0x4000		/* 16KB */
 #define PREBOOT_SIZE		0xC000		/* 48KB */
 
-#define UBOOT_SIZE		0xF0000		/* 960KB */
 #define HL_UART_BAUDRATE	115200
 #define HL_SPI_FLASH_MAX_FREQ	25000000
-#define HL_SPI_PRIMARY_BASE	0
-#define HL_SPI_SECONDARY_BASE	0x800000
 
 #ifndef __ASSEMBLY__
 
